File: admin-api/src/main/kotlin/dev/dornol/ticket/admin/api/app/dto/manager/ManagerListDto.kt
package dev.dornol.ticket.admin.api.app.dto.manager

import com.querydsl.core.annotations.QueryProjection
import dev.dornol.ticket.domain.entity.company.Company
import dev.dornol.ticket.domain.entity.manager.ManagerApproval
import dev.dornol.ticket.domain.entity.manager.ManagerRole

data class ManagerListDto(
    val id: String,
    val username: String,
    val name: String,
    val phone: String,
    val email: String,
    val managerRole: ManagerRole,
    val approval: ManagerApproval,
    val company: CompanyDto?
) {

    @QueryProjection
    constructor(
        id: Long,
        username: String,
        name: String,
        phone: String,
        email: String,
        managerRole: ManagerRole,
        approval: ManagerApproval,
        company: Company?
    ) : this(
        id.toString(),
        username,
        name,
        phone,
        email,
        managerRole,
        approval,
        company?.let { CompanyDto(it.name, it.businessNumber) })

}

File: admin/src/lib/types/manager/manager-list.dto.ts
import { ManagerRole } from "@/lib/types/auth/auth";
import { Company } from "@/lib/types/company/company.dto";

export interface ManagerListDto {
  id: string;
  username: string;
  name: string;
  phone: string;
  email: string;
  managerRole: ManagerRole;
  approval: Approval;
  company: Company;
}

export interface Approval {
  approved: boolean;
  approvedDate?: Date | null;
  approvedBy?: number | null;
}